src/helpers/puppet/PuppetMouth.ts

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * PuppetMouth is a PIXI Animate MovieClip that contains frames for various mouth shapes
+ * It is set to be called with an audio file and configuration for the lipsync
+ */
+export class PuppetMouth {
+
+    mouth:PIXI.animate.MovieClip;
+    syncFlashAmp:Array<number>;
+    syncRhubarb:Array<RhubarbJSON> | RhubarbArray;
+    update:Function;
+
+    _currentDuration:number;
+    _currentSoundInstance:PIXI.sound.IMediaInstance;
+    _talking:boolean;
+
+    constructor (object:PIXI.animate.MovieClip) {
+        this.mouth = object;
+        this.rest();
+    }
+
+    rest() {
+        this.mouth.gotoAndStop("X");
+    }
+
+    setSyncFlashAmp(syncarray:Array<number>) {
+        this.syncFlashAmp = syncarray;
+        this.update = this.updateFlashAmp.bind(this);
+    }
+
+    /**
+     * 
+     * @param syncconfig 
+     */
+    setSyncArray(syncconfig:RhubarbArray) {
+        this.syncRhubarb = syncconfig;
+        this._currentDuration = syncconfig[syncconfig.length-2] as number;
+        this.update = this.updateRhubarb.bind(this);
+    } 
+
+    /**
+     * 
+     * @param syncconfig 
+     */
+   setSyncJSON(syncconfig:Array<RhubarbJSON>) {
+       this.syncRhubarb = syncconfig;
+       this._currentDuration = syncconfig[syncconfig.length -1].end;
+       this.update = this.updateJSON.bind(this);
+   }
+
+    /**
+     * 
+     * updateFlashAmp takes in a FlashAmp-style array of frames at regular intervals
+     * e.g. [0,1,3,2,0,0,0,0,0,0,3,2,3,1,0]
+     * 
+     * @param position The position of the audio file, from 0 to 1
+     */
+    updateFlashAmp(position:number) {
+        const frame = this.syncFlashAmp[parseInt("" + (position * this.syncFlashAmp.length))];
+        this.mouth.gotoAndStop(frame);
+    }
```

Comment:
no, it was out of scope for what I was doing. The Molly code contained a reference to it so I kept it in there. Do you think that we should remove it?

Reply:
I don't know that anybody uses or will be using FlashAmp, but we used to. I had it in the Molly code in case we had a tool whose output was like FlashAmp's or in case we converted - the sync code for that kind of output is simpler. But it could probably be removed.

Reply:
I will remove it from this example to simplify, since this is supposed to be rhubarb.